src/components/input/SelectNext.tsx

```diff
@@ -257,52 +257,57 @@ type MultiValueProps<T> = {
-    buttonContent?: ComponentChildren;
-    disabled?: boolean;
+type BaseSelectProps = CompositeProps & {
+  buttonContent?: ComponentChildren;
+  disabled?: boolean;
 
-    /**
-     * Whether this select should allow multi-selection or not.
-     * When this is true, the listbox is kept open when an option is selected
-     * and the value must be an array.
-     * Defaults to false.
-     */
-    multiple?: boolean;
+  /**
+   * `id` attribute for the toggle button. This is useful to associate a label
+   * with the control.
+   */
+  buttonId?: string;
 
-    /**
-     * `id` attribute for the toggle button. This is useful to associate a label
-     * with the control.
-     */
-    buttonId?: string;
+  /** Additional classes to pass to container */
+  containerClasses?: string | string[];
+  /** Additional classes to pass to toggle button */
+  buttonClasses?: string | string[];
+  /** Additional classes to pass to listbox */
+  listboxClasses?: string | string[];
 
-    /** Additional classes to pass to container */
-    containerClasses?: string | string[];
-    /** Additional classes to pass to toggle button */
-    buttonClasses?: string | string[];
-    /** Additional classes to pass to listbox */
-    listboxClasses?: string | string[];
+  /**
+   * Align the listbox to the right.
+   * Useful when the listbox is bigger than the toggle button and this component
+   * is rendered next to the right side of the page/container.
+   * Defaults to false.
+   */
+  right?: boolean;
 
-    /**
-     * Align the listbox to the right.
-     * Useful when the listbox is bigger than the toggle button and this component
-     * is rendered next to the right side of the page/container.
-     * Defaults to false.
-     */
-    right?: boolean;
+  'aria-label'?: string;
+  'aria-labelledby'?: string;
 
-    'aria-label'?: string;
-    'aria-labelledby'?: string;
+  /**
+   * Used to determine if the listbox should use the popover API.
+   * Defaults to true, as long as the browser supports it.
+   */
+  listboxAsPopover?: boolean;
 
-    /**
-     * Used to determine if the listbox should use the popover API.
-     * Defaults to true, as long as the browser supports it.
-     */
-    listboxAsPopover?: boolean;
+  /** A callback passed to the listbox onScroll */
+  onListboxScroll?: JSX.HTMLAttributes<HTMLUListElement>['onScroll'];
+};
+
+export type SelectProps<T> = BaseSelectProps & SingleValueProps<T>;
 
-    /** A callback passed to the listbox onScroll */
-    onListboxScroll?: JSX.HTMLAttributes<HTMLUListElement>['onScroll'];
-  };
+export type MultiSelectProps<T> = BaseSelectProps & MultiValueProps<T>;
+
+export type SelectNextProps<T> = (SelectProps<T> | MultiSelectProps<T>) & {
+  /**
+   * Whether this select should allow multi-selection or not.
+   * When this is true, the listbox is kept open when an option is selected
+   * and the value must be an array.
+   * Defaults to false.
+   */
+  multiple?: boolean;
```

Comment:
I considered moving this to `SingleValueProps` and `MultiValueProps`, but that breaks other things, so I decided to leave it as is.

With `Select` and `MultiSelect`, this prop's value will be implicit and consumers will not have to provide it, so it's not too bad.